tests/components/insteon/test_api_config.py

```diff
@@ -0,0 +1,391 @@
+"""Test the Insteon APIs for configuring the integration."""
+
+from unittest.mock import patch
+
+from homeassistant.components.insteon.api.device import ID, TYPE
+from homeassistant.components.insteon.const import (
+    CONF_HUB_VERSION,
+    CONF_OVERRIDE,
+    CONF_X10,
+)
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    MOCK_DEVICE,
+    MOCK_HOSTNAME,
+    MOCK_USER_INPUT_HUB_V1,
+    MOCK_USER_INPUT_HUB_V2,
+    MOCK_USER_INPUT_PLM,
+)
+from .mock_connection import mock_failed_connection, mock_successful_connection
+from .mock_setup import async_mock_setup
+
+from tests.typing import WebSocketGenerator
+
+
+class MockProtocol:
+    """A mock Insteon protocol object."""
+
+    connected = True
+
+
+async def test_get_config(
+    hass: HomeAssistant, hass_ws_client: WebSocketGenerator
+) -> None:
+    """Test getting the Insteon configuration."""
+
+    ws_client, _, _, _ = await async_mock_setup(hass, hass_ws_client)
+    await ws_client.send_json({ID: 2, TYPE: "insteon/config/get"})
+    msg = await ws_client.receive_json()
+    result = msg["result"]
+
+    assert result["modem_config"] == {"device": MOCK_DEVICE}
+
+
+async def test_get_modem_schema_plm(
+    hass: HomeAssistant, hass_ws_client: WebSocketGenerator
+) -> None:
+    """Test getting the Insteon PLM modem configuration schema."""
+
+    ws_client, _, _, _ = await async_mock_setup(hass, hass_ws_client)
+    await ws_client.send_json({ID: 2, TYPE: "insteon/config/get_modem_schema"})
+    msg = await ws_client.receive_json()
+    result = msg["result"][0]
+
+    assert result["default"] == MOCK_DEVICE
+    assert result["name"] == "device"
+    assert result["required"]
+
+
+async def test_get_modem_schema_hub(
+    hass: HomeAssistant, hass_ws_client: WebSocketGenerator
+) -> None:
+    """Test getting the Insteon PLM modem configuration schema."""
+
+    ws_client, devices, _, _ = await async_mock_setup(
+        hass,
+        hass_ws_client,
+        config_data={**MOCK_USER_INPUT_HUB_V2, CONF_HUB_VERSION: 2},
+    )
+    await ws_client.send_json({ID: 2, TYPE: "insteon/config/get_modem_schema"})
+    msg = await ws_client.receive_json()
+    result = msg["result"][0]
+
+    assert result["default"] == MOCK_HOSTNAME
+    assert result["name"] == "host"
+    assert result["required"]
+
+
+async def test_update_modem_config_plm(
+    hass: HomeAssistant, hass_ws_client: WebSocketGenerator
+) -> None:
+    """Test getting the Insteon PLM modem configuration schema."""
+
+    ws_client, mock_devices, _, _ = await async_mock_setup(
+        hass,
+        hass_ws_client,
+    )
+    with (
+        patch(
+            "homeassistant.components.insteon.api.config.async_connect",
+            new=mock_successful_connection,
+        ),
+        patch("homeassistant.components.insteon.api.config.devices", mock_devices),
+        patch("homeassistant.components.insteon.api.config.async_close"),
+    ):
+        await ws_client.send_json(
```

Comment:
We can use the `send_json_auto_id` helper so we don't need to include an id count.